src/wp-includes/rest-api/endpoints/class-wp-rest-block-navigation-areas-controller.php

```diff
@@ -0,0 +1,285 @@
+<?php
+/**
+ * REST API: WP_REST_Block_Navigation_Areas_Controller class
+ *
+ * @subpackage REST_API
+ * @package    WordPress
+ */
+
+/**
+ * Core class used to access block navigation areas via the REST API.
+ *
+ * @see   WP_REST_Controller
+ */
+class WP_REST_Block_Navigation_Areas_Controller extends WP_REST_Controller {
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		$this->namespace = 'wp/v2';
+		$this->rest_base = 'block-navigation-areas';
+	}
+
+	/**
+	 * Registers the routes for the objects of the controller.
+	 *
+	 * @see   register_rest_route()
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base,
+			array(
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_items' ),
+					'permission_callback' => array( $this, 'get_items_permissions_check' ),
+					'args'                => $this->get_collection_params(),
+				),
+				'schema' => array( $this, 'get_public_item_schema' ),
+				'allow_batch' => array( 'v1' => true ),
```

Comment:
`allow_batch` doesn't make sense here since this is a `GET` endpoint which we don't support batching for. Instead this should be in the route registration for the second set of routes.